### ISSUE TITLE
Firebase remote config - Can not get key list from default value dictionary

### ISSUE THREAD

**Issue by takaaptech** [firebase/quickstart-unity#62]
I try to get key, value list from defaults dictionary, in editor mode it works. But on real device (android device). It print nothing from for each loop.
Example MonoBehaviour as bellow:

```
using System;
using System.Threading.Tasks;
using Firebase.RemoteConfig;
using OneSoftGame.Tools;
using System.Diagnostics;
using System.Collections;
using UnityEngine;

public class RMManager : MonoBehaviour
{

    Firebase.DependencyStatus dependencyStatus = Firebase.DependencyStatus.UnavailableOther;

    /// <summary>
    /// Awake is called when the script instance is being loaded.
    /// </summary>
    void Start()
    {
        dependencyStatus = Firebase.FirebaseApp.CheckDependencies();
        if (dependencyStatus != Firebase.DependencyStatus.Available)
        {
            Firebase.FirebaseApp.FixDependenciesAsync().ContinueWith(task =>
            {
                dependencyStatus = Firebase.FirebaseApp.CheckDependencies();
                if (dependencyStatus == Firebase.DependencyStatus.Available)
                {
                    print("RemoteConfig configured and ready!");
					InitFB();
                }
                else
                {
                    print(
                        "Could not resolve all Firebase dependencies: " + dependencyStatus);
                }
            });
        }
        else
        {
            print("RemoteConfig configured and ready!");
			InitFB();
        }
    }

    void InitFB()
    {
        print("Start InitFB");
		ListDefaultValues();
        //Test("key1");
        //Test("key2");
        SetDefaultValues("key1", "value1");
        //Test("key1");
        //Test("key2");
        ListDefaultValues();
        SetDefaultValues("key2", "value2");
        //Test("key1");
        //Test("key2");
        ListDefaultValues();

    }
    void Test(string key)
    {
           print("Start Test key:"+ key);
           var item = Firebase.RemoteConfig.FirebaseRemoteConfig.GetValue(key);
            DebugFirebaseConfigValue(item);
    }

    void SetDefaultValues(string key, object value)
    {
        print("Start SetDefaultValues for key:"+ key);
        System.Collections.Generic.Dictionary<string, object> defaults =
            new System.Collections.Generic.Dictionary<string, object>();
        defaults.Add(key, value);
        Firebase.RemoteConfig.FirebaseRemoteConfig.SetDefaults(defaults);
        ListDefaultValues();
    }

    private void ListDefaultValues()
    {
        print("Start ListDefaultValues GetKeysByPrefix ++++");
        var keys = Firebase.RemoteConfig.FirebaseRemoteConfig.GetKeysByPrefix("","");


        foreach (var key in keys)
        {
            print("ListDefaultValues:" + key);
            var item = Firebase.RemoteConfig.FirebaseRemoteConfig.GetValue(key);
            DebugFirebaseConfigValue(item);
        }
        print("Start ListDefaultValues GetKeys ++++");
        keys = Firebase.RemoteConfig.FirebaseRemoteConfig.GetKeys("");
        
        foreach (var key in keys)
        {
            print("ListDefaultValues:" + key);
            var item = Firebase.RemoteConfig.FirebaseRemoteConfig.GetValue(key);
            DebugFirebaseConfigValue(item);
        }

        print("Start ListDefaultValues .Keys ++++");
        keys = Firebase.RemoteConfig.FirebaseRemoteConfig.Keys;
        
        foreach (var key in keys)
        {
            print("ListDefaultValues:" + key);
            var item = Firebase.RemoteConfig.FirebaseRemoteConfig.GetValue(key);
            DebugFirebaseConfigValue(item);
        }


        print("End ListDefaultValues ++++");
    }

    public void DebugFirebaseConfigValue(ConfigValue item)
    {
        try
        {
            print("Debug ConfigValue Start -----------------");
            print("Source:" + item.Source);
            print("StringValue:" + item.StringValue);
            print("Debug ConfigValue End -----------------");
        }
        catch (Exception ex)
        {
            print("DebugFirebaseConfigValue Exception:" + ex.Message);
        }

    }
}
```

**Comment by takaaptech** [comment]
I just build TestApp from this repo in to android devices, and when press Display All Keys, this is list only key have on server (Not defaults key set from client)

**Comment by a-maurice** [comment]
Sorry for the delay in response.  Unfortunately, that's an underlying bug, where it is only looking at server keys, and it may be a bit before we can get a fix in.

In the meantime, if you need to be able to get the keys from SetDefaults, I'd recommend preserving them yourself in a List when you perform the call to SetDefaults.  If you need more details on exactly how that might work, let me know.

Hopefully that helps, and thanks for finding this issue!

**Comment by frankkienl** [comment]
For people using Remote Config, but with the Unity:
https://stackoverflow.com/a/44588535/1398449

(There is no bug-tracker for "normal Android" Firebase sdk)

**Comment by stewartmiles** [comment]
We have this fix linked up for the next minor release, 4.1.x.  We'll post an update when it's ready.

**Comment by stewartmiles** [comment]
This issue is now resolved in the 4.1.0 release of the Firebase Unity SDK.  For more information, see
https://firebase.google.com/support/release-notes/unity#4.1.0

Cheers,
Stewart
